osflow/boards/index.mk

```diff
@@ -115,3 +115,17 @@ PNRFLAGS    ?= --85k --freq 25 --package CABGA381 --ignore-loops --timing-allow-
 IMPL        ?= neorv32_$(BOARD)_$(ID)
 
 endif
+
+ifeq ($(BOARD),nexys_a7)
+$(info Setting constraints and implementation args for BOARD nexys_a7)
+
+SYNTH_XILINX_OPTIONS = -flatten -abc9 -arch xc7
+
+DEVICE_SERIES = xilinx
+DEVICE_PART   = xc7a100tcsg324-1
+
+CONSTRAINTS ?= $(PCF_PATH)/$(BOARD).xdc
```

Comment:
I suggest to pick the constraints from submodule `constraints`, rather than adding a local XDC file. See [hdl/constraints: board/Nexys4](https://github.com/hdl/constraints/tree/main/board/Nexys4) and [hdl/constraints: board/Nexys4DDR](https://github.com/hdl/constraints/tree/main/board/Nexys4DDR).

Reply:
Ohh, I wasn't aware ot that. I did all my work in a branch of the main neorv32-project (without this contraints-directory) and just moved it when I found this repo.

Reply:
Yeah, it was planned since some months ago, but we did not apply it until very few days ago. It's ok.

Moreover, in the Nexys4 and Nexys4DDR subdirs of hdl/constraints, there is no "single file" including all of them. There are partial constraints files, meant to be used with tools that support them. I don't know if nextpnr-xilinx supports that. Otherwise, there are two alternatives:

- Make a PR in hdl/constraints to merge all of them. See, for instance, `constraints.pcf` in https://github.com/hdl/constraints/tree/main/board/IceStick.
- Use `cat` in the makefiles here, to merge the wanted partial constraints right before calling nextpnr-xilinx.